src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -146,13 +145,83 @@ private static bool ApplyChangesFromTC31(JObject sarifLog)
             return modifiedLog;
         }
 
+        private static bool MoveRulesMetadataAndConfiguration(JObject run)
+        {
+            if (!(run["resources"] is JObject resources))
```

Comment:
>! [](start = 16, length = 1)

Here's a case where you _did_ use the new pattern in a negative case, and I think it's just fine.

Reply:
oh right, turns out i did convert all cases. only remaining ones relate to use of string type...

---
In reply to: [255311429](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255311429) [](ancestors = 255311429)